Sources/XcodesKit/RuntimeList.swift

```diff
@@ -0,0 +1,75 @@
+import PromiseKit
+import Foundation
+import Version
+import Path
+
+public class RuntimeList {
+    public init() {
+    }
+
+    public func printAvailableRuntimes() -> Promise<Void> {
+        when(fulfilled: downloadableRuntimes(), installedRuntimes())
+            .done { downloadables, installed in
+                var installed = installed
+                for (platform, downloadables) in Dictionary(grouping: downloadables, by: \.platform).sorted(\.key.order) {
+                    Current.logging.log("-- \(platform.shortName) --")
+                    for downloadable in downloadables.sorted(\.version) {
+                        let matchingInstalledRuntimes = installed.remove { $0.build == downloadable.simulatorVersion.buildUpdate }
+                        let name = downloadable.platform.shortName + " \(downloadable.simulatorVersion.version)"
+                        if !matchingInstalledRuntimes.isEmpty {
+                            for matchingInstalledRuntime in matchingInstalledRuntimes {
+                                if matchingInstalledRuntime.kind == .legacyDownload {
+                                    Current.logging.log(name + " (Downloaded)")
+                                } else if matchingInstalledRuntime.kind == .bundled {
+                                    Current.logging.log(name + " (Bundled with selected Xcode)")
+                                }
+                            }
+                        } else {
+                            Current.logging.log(name)
+                        }
+                    }
+                }
+            }
+    }
+
+    func downloadableRuntimes() -> Promise<[DownloadableRuntime]> {
+        return firstly { () -> Promise<(data: Data, response: URLResponse)> in
+            Current.network.dataTask(with: URLRequest.runtimes)
+        }
+        .map { (data, response) -> [DownloadableRuntime] in
+            let response = try PropertyListDecoder().decode(DownloadableRuntimesResponse.self, from: data)
+            return response.downloadables.filter { downloadable in
+                !downloadable.name.lowercased().contains("beta")
```

Comment:
Would there be any reason why we _couldn't_ install a beta simulator?

Reply:
I don't think so, I just did because Xcode doesn't list them in the UI. 
I removed the filtering 👍🏼

Reply:
Sorry @StevenMagdy - looking at how many betas are returned. i'm wondering if we keep this in, and have the additional option of `--include-betas` that will allow all the files? I'm ok with an additional PR later if that is simpler.

Reply:
@MattKiazyk Great suggestion. Added 👍🏼